### PR TITLE
Hexa8 cantilever beam

### PR DESCRIPTION
**Pull request by christmour**
[Hexa8 Cantilever Beam.pdf](https://github.com/isaar/MSolve/files/479034/Hexa8.Cantilever.Beam.pdf)

### CODE REVIEW

File: ISAAR.MSolve.SamplesConsole/HexaSimpleCantileverBeam.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                model.NodesDictionary[nodeID].Constraints.Add(DOFType.Z);
+
+                nodeID++;
+            }
+
+            for (int i = 0; i < 4; i++)
+            {
+                for (int k = 0; k < 4; k++)
+                {
+                    if (nodeID % 2 == 0)
```
Comment (georgestavroulakis) [inline ISAAR.MSolve.SamplesConsole/HexaSimpleCantileverBeam.cs:40]: This `if...else` block can be written in one line, please revise as follows:

`model.NodesDictionary.Add(nodeID, new Node() { ID = nodeID, X = startX + 0.25 * (i + 1), Y = (nodeID % 2 == 0 ? startY + 0.25 : startY), Z = startZ + 0.25 * (k / 2) });`

File: ISAAR.MSolve.SamplesConsole/HexaSimpleCantileverBeam.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        public static void MakeCantileverBeam(Model model, double startX, double startY, double startZ, int startNodeID, int startElementID, int subdomainID)
+
+        {
+
+            int nodeID = startNodeID;
+
+            for (int j = 0; j < 4; j++)
+            {
+                if (nodeID % 2 == 0)
```
Comment (georgestavroulakis) [inline ISAAR.MSolve.SamplesConsole/HexaSimpleCantileverBeam.cs:21]: This `if...else` block can be written in one line, please revise as follows:

`model.NodesDictionary.Add(nodeID, new Node() { ID = nodeID, X = startX, Y = (nodeID % 2 == 0 ? startY + 0.25 : startY), Z = startZ + 0.25 * (j / 2) });
`

File: ISAAR.MSolve.SamplesConsole/Program.cs [hunk 3]
```diff
@@ -56,10 +56,39 @@ private static void SolveBuildingInNoSoilSmallDynamic()
...
 
+        private static void SolveHexaCantileverBeam()
+        {
+            VectorExtensions.AssignTotalAffinityCount();
+            Model model = new Model();
+            model.SubdomainsDictionary.Add(1, new Subdomain() { ID = 1 });
+                       
+            HexaSimpleCantileverBeam.MakeCantileverBeam(model, 0, 0, 0, model.NodesDictionary.Count + 1, model.ElementsDictionary.Count + 1, 1);
+
+            model.Loads.Add(new Load() { Amount = -0.25, Node = model.Nodes[16], DOF = DOFType.Z });
```
Comment (georgestavroulakis) [inline ISAAR.MSolve.SamplesConsole/Program.cs:67]: Replace all loading lines with the following for loop:

```
for (int i = 16; i < 20; i++)
  model.Loads.Add(new Load() { Amount = -0.25, Node = model.Nodes[i], DOF = DOFType.Z });

```

**Review by GoatOfRafin — CHANGES_REQUESTED** [review]
Small revisions are required in order to obtain a more concise source code.

**Comment by GoatOfRafin** [comment]
Please re submit pull request at the development branch

**Pull request closed without merging**